packages/validator/src/validator.ts

```diff
@@ -0,0 +1,148 @@
+  }
+
+  try {
+    template = JSON.parse(template);
+  } catch (e) {
+    return { valid: false, messages: [logger.error(`${templatePath} is not valid JSON: ${e}`)] };
+  }
+
+  try {
+    specs = fs.readFileSync(path.resolve(specsPath), 'utf-8');
+  } catch (e) {
+    return { valid: false, messages: [logger.error(`Unable to read file ${path.resolve(specsPath)}`)] };
+  }
+
+  try {
+    specs = JSON.parse(specs);
+  } catch (e) {
+    return { valid: false, messages: [logger.error(`${specsPath} is not valid JSON: ${e}`)] };
+  }
+
+  const split = templatePath.split('/');
+
+  return validateJson(specs, template, split.slice(0, split.length - 1).join('/') + '/', returnJson);
+}
+
+/**
+ * Validates JSON specification with known template
+ * @param specs - specification to validate
+ * @param templateName - name of the template (ois, config...)
+ * @param returnJson - parsed JSON specification will be returned
+ */
+export function validateJsonWithTemplate(specs: object, templateName: string | undefined, returnJson = false): Result {
+  if (!templateName) {
+    return { valid: false, messages: [logger.error('Specification and template file must be provided')] };
+  }
+
+  const messages: Log[] = [];
+
+  const templatePath = utils.getPath(templates[templateName.toLowerCase() as keyof typeof templates], messages);
+
+  if (messages.length) {
+    return { valid: false, messages };
+  }
+
+  let template;
+
+  try {
+    template = fs.readFileSync(templatePath, 'utf-8');
+  } catch (e) {
+    return { valid: false, messages: [logger.error(`Unable to read file ${templatePath}`)] };
+  }
+
+  try {
+    template = JSON.parse(template);
+  } catch (e) {
+    return { valid: false, messages: [logger.error(`${templatePath} is not valid JSON: ${e}`)] };
+  }
+
+  const split = templatePath.split('/');
+
+  return validateJson(specs, template, split.slice(0, split.length - 1).join('/') + '/', returnJson);
+}
+
+/**
+ * Validates specification on provided path with known template
+ * @param specsPath - path to specification that will be validated
+ * @param templateName - name of the template (ois, config...)
+ * @param returnJson - parsed JSON specification will be returned
+ */
+export function validateWithTemplate(
+  specsPath: string | undefined,
+  templateName: string | undefined,
+  returnJson = false
+) {
+  if (!specsPath) {
+    return { valid: false, messages: [logger.error('Specification and template file must be provided')] };
+  }
+
+  let specs;
+
+  try {
+    specs = fs.readFileSync(path.resolve(specsPath), 'utf-8');
+  } catch (e) {
+    return { valid: false, messages: [logger.error(`Unable to read file ${path.resolve(specsPath)}`)] };
+  }
+
+  try {
+    specs = JSON.parse(specs);
+  } catch (e) {
+    return { valid: false, messages: [logger.error(`${specsPath} is not valid JSON: ${e}`)] };
+  }
+
+  return validateJsonWithTemplate(specs, templateName, returnJson);
+}
```

Comment:
So, what are these two functions good for? As far as I can tell they're not used anywhere, are they?

Reply:
They become useful when validator is used as a package, check the [deployer integration PR](https://github.com/api3dao/airnode/pull/391) for example.